### ISSUE TITLE
Many-to-many: integrity constraint violation

### ISSUE THREAD

**Issue by roxblnfk** [cycle/orm#70]
If I add fields that are generated by relationships to the crosstab Entity class:
```php
/** @Entity */
class PostTag
{
    /** @Column(type="primary") */
    private ?int $id = null;
    private ?int $post_id = null;
    private ?int $tag_id = null;
}
```
And when adding fixtures i will bind both entities:
```php
$post->addTag($tag);
$tag->addPost($post);
```
Then i get an error:
![image](https://user-images.githubusercontent.com/4152481/74154890-945edd00-4c24-11ea-9968-40ea1792e595.png)

If i remove `$tag->addPost($post);` or fields `$post_id` and `$tag_id` then will be good

Relations in entities:
Entity Tag https://github.com/roxblnfk/yii-demo/blob/bug/many-to-many/src/Blog/Entity/Post.php#L90
Entity Post https://github.com/roxblnfk/yii-demo/blob/bug/many-to-many/src/Blog/Entity/Tag.php#L45

php 7.4.1

### How to reproduce
 0. `git clone https://github.com/roxblnfk/yii-demo.git` and `git checkout bug/many-to-many`
 1. Configure `config/params.php` _\(you can skip this step)_
 2. Run `composer install`
 3. Run `vendor/bin/yii serve` or start your web-server
 4. Go to index page. \
    On this step the Cycle ORM will create tables, indexes and relations automatically in the configured DB. \
    If you want to disable this behavior then comment out line with `Generator\SyncTables::class` in the `config/params.php` file. In this case you should create migrations to sync changes of entities with DB
 5. Run `vendor/bin/yii fixture/add 20` to create random data

**Comment by wolfy-j** [comment]
Can I see your schema? Did you use `@inverse`?

**Comment by roxblnfk** [comment]
```php
<?php

use Cycle\ORM\Schema;
use Cycle\ORM\Relation;

return [
    'user' => [
        Schema::DATABASE => 'default',
        Schema::TABLE => 'user',
        Schema::ENTITY => 'App\\Entity\\User',
        Schema::MAPPER => 'Yiisoft\\Yii\\Cycle\\Mapper\\TimestampedMapper',
        Schema::REPOSITORY => 'App\\Repository\\UserRepository',
        Schema::CONSTRAIN => null,
        Schema::PRIMARY_KEY => 'id',
        Schema::COLUMNS => [
            'id' => 'id',
            'token' => 'token',
            'login' => 'login',
            'passwordHash' => 'password_hash',
            'created_at' => 'created_at',
            'updated_at' => 'updated_at',
        ],
        Schema::TYPECAST => [
            'id' => 'int',
            'created_at' => 'datetime',
            'updated_at' => 'datetime',
        ],
        Schema::RELATIONS => [
            'posts' => [
                Relation::TYPE => Relation::HAS_MANY,
                Relation::TARGET => 'post',
                Relation::LOAD => Relation::LOAD_PROMISE,
                Relation::SCHEMA => [
                    Relation::CASCADE => true,
                    Relation::NULLABLE => false,
                    Relation::WHERE => [],
                    Relation::INNER_KEY => 'id',
                    Relation::OUTER_KEY => 'user_id',
                ],
            ],
            'comments' => [
                Relation::TYPE => Relation::HAS_MANY,
                Relation::TARGET => 'comment',
                Relation::LOAD => Relation::LOAD_PROMISE,
                Relation::SCHEMA => [
                    Relation::CASCADE => true,
                    Relation::NULLABLE => false,
                    Relation::WHERE => [],
                    Relation::INNER_KEY => 'id',
                    Relation::OUTER_KEY => 'user_id',
                ],
            ],
        ],
    ],
    'comment' => [
        Schema::DATABASE => 'default',
        Schema::TABLE => 'comment',
        Schema::ENTITY => 'App\\Blog\\Entity\\Comment',
        Schema::MAPPER => 'App\\Blog\\Comment\\CommentMapper',
        Schema::REPOSITORY => 'Cycle\\ORM\\Select\\Repository',
        Schema::CONSTRAIN => 'App\\Blog\\Comment\\Scope\\PublicScope',
        Schema::PRIMARY_KEY => 'id',
        Schema::COLUMNS => [
            'id' => 'id',
            'public' => 'public',
            'content' => 'content',
            'created_at' => 'created_at',
            'updated_at' => 'updated_at',
            'published_at' => 'published_at',
            'deleted_at' => 'deleted_at',
            'user_id' => 'user_id',
            'post_id' => 'post_id',
        ],
        Schema::TYPECAST => [
            'id' => 'int',
            'public' => 'bool',
            'created_at' => 'datetime',
            'updated_at' => 'datetime',
            'published_at' => 'datetime',
            'deleted_at' => 'datetime',
            'user_id' => 'int',
            'post_id' => 'int',
        ],
        Schema::RELATIONS => [
            'user' => [
                Relation::TYPE => Relation::BELONGS_TO,
                Relation::TARGET => 'user',
                Relation::LOAD => Relation::LOAD_EAGER,
                Relation::SCHEMA => [
                    Relation::CASCADE => true,
                    Relation::NULLABLE => false,
                    Relation::INNER_KEY => 'user_id',
                    Relation::OUTER_KEY => 'id',
                ],
            ],
            'post' => [
                Relation::TYPE => Relation::BELONGS_TO,
                Relation::TARGET => 'post',
                Relation::LOAD => Relation::LOAD_PROMISE,
                Relation::SCHEMA => [
                    Relation::CASCADE => true,
                    Relation::NULLABLE => false,
                    Relation::INNER_KEY => 'post_id',
                    Relation::OUTER_KEY => 'id',
                ],
            ],
        ],
    ],
    'post' => [
        Schema::DATABASE => 'default',
        Schema::TABLE => 'post',
        Schema::ENTITY => 'App\\Blog\\Entity\\Post',
        Schema::MAPPER => 'App\\Blog\\Post\\PostMapper',
        Schema::REPOSITORY => 'App\\Blog\\Post\\PostRepository',
        Schema::CONSTRAIN => 'App\\Blog\\Post\\Scope\\PublicScope',
        Schema::PRIMARY_KEY => 'id',
        Schema::COLUMNS => [
            'id' => 'id',
            'slug' => 'slug',
            'title' => 'title',
            'public' => 'public',
            'content' => 'content',
            'created_at' => 'created_at',
            'updated_at' => 'updated_at',
            'published_at' => 'published_at',
            'deleted_at' => 'deleted_at',
            'user_id' => 'user_id',
        ],
        Schema::TYPECAST => [
            'id' => 'int',
            'public' => 'bool',
            'created_at' => 'datetime',
            'updated_at' => 'datetime',
            'published_at' => 'datetime',
            'deleted_at' => 'datetime',
            'user_id' => 'int',
        ],
        Schema::RELATIONS => [
            'user' => [
                Relation::TYPE => Relation::BELONGS_TO,
                Relation::TARGET => 'user',
                Relation::LOAD => Relation::LOAD_PROMISE,
                Relation::SCHEMA => [
                    Relation::CASCADE => true,
                    Relation::NULLABLE => false,
                    Relation::INNER_KEY => 'user_id',
                    Relation::OUTER_KEY => 'id',
                ],
            ],
            'tags' => [
                Relation::TYPE => Relation::MANY_TO_MANY,
                Relation::TARGET => 'tag',
                Relation::LOAD => Relation::LOAD_PROMISE,
                Relation::SCHEMA => [
                    Relation::CASCADE => true,
                    Relation::NULLABLE => false,
                    Relation::WHERE => [],
                    Relation::INNER_KEY => 'id',
                    Relation::OUTER_KEY => 'id',
                    Relation::THROUGH_ENTITY => 'postTag',
                    Relation::THROUGH_INNER_KEY => 'post_id',
                    Relation::THROUGH_OUTER_KEY => 'tag_id',
                    Relation::THROUGH_WHERE => [],
                ],
            ],
            'comments' => [
                Relation::TYPE => Relation::HAS_MANY,
                Relation::TARGET => 'comment',
                Relation::LOAD => Relation::LOAD_PROMISE,
                Relation::SCHEMA => [
                    Relation::CASCADE => true,
                    Relation::NULLABLE => false,
                    Relation::WHERE => [],
                    Relation::INNER_KEY => 'id',
                    Relation::OUTER_KEY => 'post_id',
                ],
            ],
        ],
    ],
    'postTag' => [
        Schema::DATABASE => 'default',
        Schema::TABLE => 'post_tag',
        Schema::ENTITY => 'App\\Blog\\Entity\\PostTag',
        Schema::MAPPER => 'Cycle\\ORM\\Mapper\\Mapper',
        Schema::REPOSITORY => 'Cycle\\ORM\\Select\\Repository',
        Schema::CONSTRAIN => null,
        Schema::PRIMARY_KEY => 'id',
        Schema::COLUMNS => [
            'id' => 'id',
            'post_id' => 'post_id',
            'tag_id' => 'tag_id',
        ],
        Schema::TYPECAST => [
            'id' => 'int',
            'post_id' => 'int',
            'tag_id' => 'int',
        ],
        Schema::RELATIONS => [],
    ],
    'tag' => [
        Schema::DATABASE => 'default',
        Schema::TABLE => 'tag',
        Schema::ENTITY => 'App\\Blog\\Entity\\Tag',
        Schema::MAPPER => 'App\\Blog\\Tag\\TagMapper',
        Schema::REPOSITORY => 'App\\Blog\\Tag\\TagRepository',
        Schema::CONSTRAIN => null,
        Schema::PRIMARY_KEY => 'id',
        Schema::COLUMNS => [
            'id' => 'id',
            'label' => 'label',
            'created_at' => 'created_at',
        ],
        Schema::TYPECAST => [
            'id' => 'int',
            'created_at' => 'datetime',
        ],
        Schema::RELATIONS => [
            'posts' => [
                Relation::TYPE => Relation::MANY_TO_MANY,
                Relation::TARGET => 'post',
                Relation::LOAD => Relation::LOAD_PROMISE,
                Relation::SCHEMA => [
                    Relation::CASCADE => true,
                    Relation::NULLABLE => false,
                    Relation::WHERE => [],
                    Relation::INNER_KEY => 'id',
                    Relation::OUTER_KEY => 'id',
                    Relation::THROUGH_ENTITY => 'postTag',
                    Relation::THROUGH_INNER_KEY => 'tag_id',
                    Relation::THROUGH_OUTER_KEY => 'post_id',
                    Relation::THROUGH_WHERE => [],
                ],
            ],
        ],
    ],
];
```

**Comment by wolfy-j** [comment]
Thank you.

**Comment by roxblnfk** [comment]
If project from my repo now not works correctly, you can install current version of `yiisoft/yii-demo` and uncomment [this line](https://github.com/yiisoft/yii-demo/blob/b5c9cf67f86b38fe324bb8eb34422817fc617d82/src/Command/Fixture/AddCommand.php#L136)

**Comment by wolfy-j** [comment]
Weird, this test works: https://github.com/cycle/orm/blob/master/tests/ORM/CyclicManyToManyTest.php#L250

**Comment by wolfy-j** [comment]
Fixed in v1.3.3
